### DISCUSSION TITLE
How to read on-time / off-time values & chart ?

### DISCUSSION THREAD

**Discussion by winka576** [jmcollin78/versatile_thermostat#457] [Q&A]
Hi community, 

I am wondering how to read the time value and the chart for the on-time / off-time period
<img width="376" alt="Capture d’écran 2024-05-03 à 11 33 16" src="https://github.com/jmcollin78/versatile_thermostat/assets/167139563/9435aeb0-a5b2-4038-8195-2051437abac4">
Unless I am mistaken, the Y axis is in Second but yet I do not understand how to interprete this chart. 
On top of that, what is the unit in the card below representing On & Off time (0 & 5:00) ? and how to interprete these values overall ?

thanks for your precious help, 
Cheers, 
Kevin

**Comment by jmcollin78** [comment]
Hello @winka576 ,

In a 'over switch' VTherm, which is the case here I guess, the regulation is based on a cycle (5 min in your example).
At cycle, the VTherm calculate the time the heater should be in 'on' and the time the heater should be in 'off'. You have the algorithm it the README for information.

Of course time_on + time_off = cycle duration.

So what you see here is the time_on value in second.
This is not the most significant value because it is dependant of your cycle value. You should rather use the `percent_on` value which gives the percentage of the cycle in the 'on' state. I see you have it in the card below and it is at 0% (don't heat).

100% means that the heater will always be 'on'.

A good value in a stable case is about 20%-30%.

**Reply by winka576** [reply]
thanks Jean-Marc @jmcollin78 !
It definitely makes more sense now. 
Yet, I have adjusted the graph to look at the percentage instead as per your advice but as you can see in the below screenshot, cycle are often going way over 30%. 
<img width="410" alt="Capture d’écran 2024-05-08 à 13 32 15" src="https://github.com/jmcollin78/versatile_thermostat/assets/167139563/13de34ff-0d8b-4b05-9abf-d302fd691c4e">
what could I do to optimise the Vtherm settings ? 
thanks in advance, 
Kevin

**Reply by jmcollin78** [reply]
> cycle are often going way over 30%.

This is not mandatory an issue. What it important is that in stable conditions, the percent is not too high but if it is really cold outside or if you need more heating, the percent can up to 100% during the first heating phase.

If you want to optimize your regulation, you have many clues in the README on how to do that:
https://github.com/jmcollin78/versatile_thermostat/blob/main/README-fr.md#exemples-de-r%C3%A9glage

Et si tu veux que je regarde, il me faut des courbes comme ça : https://github.com/jmcollin78/versatile_thermostat/blob/main/README-fr.md#toujours-mieux-avec-plotly-pour-r%C3%A9gler-votre-thermostat
